### PR TITLE
default $ref resolver alters schema

### PR DESCRIPTION
**Pull request by EdVinyard**
The default `$ref` resolver previously replaced `{ $ref: "#xyz" }` with `"#xyz"`.  I _think_ the default `$ref` resolver was intended to be a no-op.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,6099 +1,8 @@
 {
   "name": "@stoplight/json-schema-merge-allof",
   "version": "0.7.7",
-  "lockfileVersion": 2,
+  "lockfileVersion": 1,
```
Comment (P0lip) [inline package-lock.json]: let's revert this change.
We should be using the new lockfile format. You probably have an outdated version of `npm`

File: src/index.js [hunk 2]
```diff
@@ -257,8 +257,8 @@ function createRequiredMetaArray(arr) {
   return {required: arr}
 }
 
-function default$RefResolver(obj) {
-  return obj
+function default$RefResolver(reference) {
+  return {$ref: reference}
```
Comment (P0lip) [inline src/index.js:261]: yup, that is correct, we didn't intend to touch the schema :+1:

**Review by P0lip — COMMENTED** [review]
Besides the lockfile change, it looks good to ne.

**Review by P0lip — APPROVED** [review]

File: package-lock.json (hunk 1 above)
Comment (EdVinyard) [reply package-lock.json]: Yikes!  Thank you for catching that.  I was indeed using an outdated NPM.

~@P0lip , should I revert _all_ changes to this file, or just the `lockfileVersion` change?~

Comment (EdVinyard) [reply package-lock.json]: Nevermind, you fixed it for me!  Thanks!

Comment (P0lip) [reply package-lock.json]: No worries!
@EdVinyard ~I'm going to release a new version, 0.7.8.~ and it's out :tada: https://npmjs.com/package/@stoplight/json-schema-merge-allof
